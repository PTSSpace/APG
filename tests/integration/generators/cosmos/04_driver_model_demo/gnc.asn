Module-gnc DEFINITIONS AUTOMATIC TAGS ::= BEGIN

  IMPORTS Cfe-tlm-header FROM Module-cfe-headers
          Real-range FROM Module-simple-types;

  Gnc-data-tlm ::= SEQUENCE { -- GNC_DATA_MID gnc_DataTlm_t (0x08A0)
      header Cfe-tlm-header (WITH COMPONENTS {
      primary (WITH COMPONENTS {
        packet-version-number (0),
        packet-identification (WITH COMPONENTS {
          packet-type-is-cmd (FALSE),
          sec-hdr-flag-is-present (TRUE),
          application-process-identifier (160)
        })
      })
    }),
    payload Payload-gnc-data-tlm
  }

  Payload-gnc-data-tlm ::= SEQUENCE {
    pos Real-range,   -- ENDIANNESS(LITTLE) [m]
    vel Real-range,   -- ENDIANNESS(LITTLE) [m/s]
    acc Real-range,   -- ENDIANNESS(LITTLE) [m/s^2]
    p Real-range,     -- ENDIANNESS(LITTLE)
    i Real-range,     -- ENDIANNESS(LITTLE)
    d Real-range,     -- ENDIANNESS(LITTLE)
    thrust Real-range -- ENDIANNESS(LITTLE) [N]
  }

END
