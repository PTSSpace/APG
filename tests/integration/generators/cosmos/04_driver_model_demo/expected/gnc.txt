# This file was autogenerated from ASN.1 model.

TELEMETRY gnc Gnc-data-tlm BIG_ENDIAN "GNC_DATA_MID gnc_DataTlm_t (0x08A0)"
    APPEND_ID_ITEM      packet-version-number                 3 UINT          0 ""
    APPEND_ID_ITEM      packet-type-is-cmd                    1 UINT          0 ""
        STATE False                           0
        STATE True                            1
    APPEND_ID_ITEM      sec-hdr-flag-is-present               1 UINT          1 ""
        STATE False                           0
        STATE True                            1
    APPEND_ID_ITEM      application-process-identifier       11 UINT       0xA0 ""
    APPEND_ITEM         sequence-flags                        2 UINT            ""
        STATE continuation-packet-in-sequence 0
        STATE first-packet-in-sequence        1
        STATE last-packet-in-sequence         2
        STATE complete-packet                 3
    APPEND_ITEM         packet-sequence-count                14 UINT            ""
    APPEND_ITEM         packet-data-length                   16 UINT            "length of the packet"
    APPEND_ITEM         seconds                              32 UINT            ""
        UNITS sec sec
    APPEND_ITEM         subsecs                              16 UINT            "unit: 2**-16 sec = 0.015 ms"
    APPEND_ITEM         spare                                32 UINT            "Spares"
    APPEND_ITEM         pos                                  64 FLOAT           "" LITTLE_ENDIAN
        LIMITS DEFAULT 5 ENABLED -1e-40 -1e-40 1e+40 1e+40
        UNITS m m
    APPEND_ITEM         vel                                  64 FLOAT           "" LITTLE_ENDIAN
        LIMITS DEFAULT 5 ENABLED -1e-40 -1e-40 1e+40 1e+40
        UNITS m/s m/s
    APPEND_ITEM         acc                                  64 FLOAT           "" LITTLE_ENDIAN
        LIMITS DEFAULT 5 ENABLED -1e-40 -1e-40 1e+40 1e+40
        UNITS m/s^2 m/s^2
    APPEND_ITEM         p                                    64 FLOAT           "" LITTLE_ENDIAN
        LIMITS DEFAULT 5 ENABLED -1e-40 -1e-40 1e+40 1e+40
    APPEND_ITEM         i                                    64 FLOAT           "" LITTLE_ENDIAN
        LIMITS DEFAULT 5 ENABLED -1e-40 -1e-40 1e+40 1e+40
    APPEND_ITEM         d                                    64 FLOAT           "" LITTLE_ENDIAN
        LIMITS DEFAULT 5 ENABLED -1e-40 -1e-40 1e+40 1e+40
    APPEND_ITEM         thrust                               64 FLOAT           "" LITTLE_ENDIAN
        LIMITS DEFAULT 5 ENABLED -1e-40 -1e-40 1e+40 1e+40
        UNITS N N
